From 10756fad5306d26982e74ae3d724181fc628aecf Mon Sep 17 00:00:00 2001
From: Scott K Logan <logans@cottsay.net>
Date: Thu, 19 May 2022 16:01:45 -0700
Subject: [PATCH] Avoid unusable system state during apt install

Installing ROS 2 on a system which hasn't get upgraded the installed
packages from the versions published at Jammy release to those published
in subsequent updates can result in undesirable behavior of the apt
resolver.

The dependency solution that apt presents seems to include the removal
of critical system packages. If the user doesn't notice the misbehavior,
it could result in an unusable system.

This patch is intended to force the upgrade of the affected packages
leaving apt no choice but to install additional packages to meet the
dependency requirements rather than uninstalling any packages.
---
 debian/control | 1 +
 1 file changed, 1 insertion(+)

diff --git a/debian/control b/debian/control
index af72f1b..eac80c9 100644
--- a/debian/control
+++ b/debian/control
@@ -9,4 +9,5 @@ Standards-Version: 3.9.2
 Package: ros-humble-ros-workspace
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, 
+Conflicts: libsystemd0 (<= 249.11-0ubuntu3), libudev1 (<= 249.11-0ubuntu3)
 Description: Provides the prefix level environment files for ROS 2 packages.
-- 
2.35.3

